rainier-core/src/main/scala/com/stripe/rainier/core/RandomVariable.scala

```diff
@@ -117,7 +109,23 @@ class RandomVariable[+T](val value: T, private val targets: Set[Target]) {
     (allSamples, diagnostics)
   }
 
-  lazy val density: Real = Real.sum(targets.map(_.inlined))
+  lazy val (variables, density) = compile()
+  private def compile() = {
+    val (variables, dataFn) = Compiler.default.compileTargets(targets, true, 4)
```

Comment:
Somewhat wasteful that we always compute gradients even when using `Walkers` but I'm not going to worry about that for now.